src/modules/job-exec/job-exec.c

```diff
@@ -1176,6 +1188,47 @@ static void exception_cb (flux_t *h, flux_msg_handler_t *mh,
     }
 }
 
+static void critical_ranks_cb (flux_t *h,
+                               flux_msg_handler_t *mh,
+                               const flux_msg_t *msg,
+                               void *arg)
+{
+    struct job_exec_ctx *ctx = arg;
+    flux_jobid_t id;
+    const char *ranks;
+    struct idset *idset;
+    struct jobinfo *job;
+    uid_t userid;
+
+    if (flux_request_unpack (msg, NULL,
+                             "{s:I s:s}",
+                             "id", &id,
+                             "ranks", &ranks) < 0)
+        goto error;
+
+    if (!(job = zhashx_lookup (ctx->jobs, &id))) {
+        errno = ENOENT;
+        goto error;
+    }
+    if (flux_msg_get_userid (msg, &userid) < 0
+        || userid != job->userid) {
+        errno = EPERM;
+        goto error;
+    }
```

Comment:
Suggestion: use the higher level functions for access control, something like
```c
struct flux_msg_cred cred;

if (flux_request_unpack (msg, ...) < 0
    || flux_request_get_cred (msg, &cred) < 0)
    goto error;
// lookup job
if (flux_msg_cred_authorize (cred, job->userid) < 0)
    goto error;
```
That will let the instance owner in too, as came up elsewhere.

Reply:
Thanks I had forgotten about this one.

What about just `flux_msg_authorize(msg, job->userid)`? Is there something that makes pulling out the `cred` required here? BTW, there is no reason I can think of that the instance owner would use this RPC, but it doesn't hurt to allow it.

Reply:
Yup i think that's fine too - I just wasn't thinking about that one :-)